cvat/apps/engine/models.py

```diff
@@ -430,6 +432,26 @@ class Meta:
     def touch(self) -> None:
         self.save(update_fields=["updated_date"])
 
+class ABCModelMeta(ABCMeta, ModelBase):
+    pass
+
+class FileSystemRelatedModel(models.Model, metaclass=ABCModelMeta):
+    class Meta:
+        abstract = True
+
+    @abstractmethod
+    def get_dirname(self) -> str:
+        ...
+
+    def get_tmp_dirname(self, *, create: bool = True) -> str:
```

Comment:
Does it ever make sense to use `create=False`? Presumably if you're getting the directory path, you're going to access it.